x-pack/plugins/observability/public/pages/slo_edit/components/slo_edit_form_indicator_section.tsx

```diff
@@ -8,17 +8,38 @@
 import { EuiFormRow, EuiPanel, EuiSelect, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { CreateSLOInput } from '@kbn/slo-schema';
-import React from 'react';
-import { Controller, useFormContext } from 'react-hook-form';
+import React, { useEffect } from 'react';
+import { Controller, useFormContext, useWatch } from 'react-hook-form';
+import { isObject } from 'lodash';
 
 import { SLI_OPTIONS } from '../constants';
 import { ApmAvailabilityIndicatorTypeForm } from './apm_availability/apm_availability_indicator_type_form';
 import { ApmLatencyIndicatorTypeForm } from './apm_latency/apm_latency_indicator_type_form';
 import { CustomKqlIndicatorTypeForm } from './custom_kql/custom_kql_indicator_type_form';
+import {
+  CustomMetricIndicatorTypeForm,
+  NEW_CUSTOM_METRIC,
+} from './custom_metric/custom_metric_type_form';
 import { maxWidth } from './slo_edit_form';
 
 export function SloEditFormIndicatorSection() {
-  const { control, watch } = useFormContext<CreateSLOInput>();
+  const { control, watch, setValue } = useFormContext<CreateSLOInput>();
+
+  const indicator = useWatch({ control, name: 'indicator.type' });
+
+  // Set the defaults for the form when switching between different indicators
+  useEffect(() => {
+    if (indicator === 'sli.metric.custom') {
+      setValue('indicator.params.good.metrics', [NEW_CUSTOM_METRIC]);
+      setValue('indicator.params.good.equation', 'A');
+      setValue('indicator.params.total.metrics', [NEW_CUSTOM_METRIC]);
+      setValue('indicator.params.total.equation', 'A');
+    }
+    if (indicator === 'sli.kql.custom' && isObject(watch('indicator.params.good'))) {
```

Comment:
I needed to do this for the `sli.kql.custom` and `sli.metric.custom` because both indicators use `good` and `total` as part of their params. For the `sli.metric.custom` I needed to set the default values.